### ISSUE TITLE
String type characters are converted to Boolean type characters

### ISSUE THREAD

**Issue by yu40ta** [aws-cloudformation/rain#480]
I am experiencing a phenomenon where source code that is recognized as "String" when deployed in the CloudFormation console is mistakenly converted to "Boolean" when deployed by rain.

For example, the following source code.

```yaml
  EC2StopSchedule:
    Type: AWS::Scheduler::Schedule
    Properties:
      Name: ec2-auto-stop
      Description: "Scheduler to auto stop EC2 instances"
      ScheduleExpression: "cron(0 19 * * ? *)"
      ScheduleExpressionTimezone: Asia/Tokyo
      FlexibleTimeWindow:
        Mode: "OFF" # this
      State: ENABLED
```

Below I paste a capture of the execution result in console and in rain.

## console
![image](https://github.com/user-attachments/assets/564a987c-cd6d-4a19-8947-17eccfd7c8fc)

## rain
![image](https://github.com/user-attachments/assets/49b5453b-590a-403e-bca2-5f1d37b84875)

Is this the expected behavior?
It appears that the double quotes have been removed for other properties except the FlexibleTimeWindow property too. (ex: Description property)
Unless this problem is resolved, templates that specify parameters with strings that can be Boolean will not be deployable.

**Comment by ericzbeard** [comment]
YAML doesn't require strings to be quoted. This might be related to #316, #319, and #329. 

#329 added this:

`To maintain original double quotes around values that require them, you can now add --node-style original to the deploy, pkg, and fmt commands.`

**Comment by richard-grainger-veeva** [comment]
@ericzbeard The issue is more subtle. If stripping quotes from a string like `"ON"` transforms it into a YAML boolean, I don't thin Rain should strip them, even if it would normally do so. Changing a data type is certainly unexpected behaviour I would suggest.

**Comment by leinardi** [comment]
@yu40ta this issue was fixed with #654

**Comment by evilneuro** [comment]
I'm seeing the same thing while using a JSON template, and rain v1.23.0. I'm updating an existing stack using `rain` that was previously deployed using the AWS CLI. The `FlexibleTimeWindow` `Mode` property value is being changed from `"OFF"` in my template to `"false"` at deployment.

Rain version check:
```shell
$ rain --version
Rain v1.23.0 linux/amd64
```

The logical resource:
```json
"EventBridgeSchedule": {
	"Type": "AWS::Scheduler::Schedule",
	"Properties": {
		"Name": "Name",
		"Description": "Description",
		"State": "DISABLED",
		"GroupName": "default",
		"ScheduleExpression": "rate(1 minutes)",
		"ScheduleExpressionTimezone": "UTC",
		"FlexibleTimeWindow": {
			"Mode": "OFF"
		},
		"Target": {
			"Arn": { "Fn::GetAtt": [ "Lambda", "Arn" ] },
			"RoleArn": { "Fn::GetAtt": [ "Role", "Arn" ] },
			"RetryPolicy": {
				"MaximumEventAgeInSeconds": 86400,
				"MaximumRetryAttempts": 100
			}
		}
	}
}
```

The context changes in the changeset:
```json
"beforeContext": "{\"Properties\":{\"GroupName\":\"default\",\"ScheduleExpression\":\"rate(1 minutes)\",\"Target\":{\"Arn\":\"LAMBDA_ARN\",\"RetryPolicy\":{\"MaximumRetryAttempts\":\"100\",\"MaximumEventAgeInSeconds\":\"86400\"},\"RoleArn\":\"ROLE_ARN\"},\"Description\":\"\",\"State\":\"DISABLED\",\"FlexibleTimeWindow\":{\"Mode\":\"OFF\"},\"ScheduleExpressionTimezone\":\"UTC\",\"Name\":\"RESOURCE_NAME\"}}",
"afterContext": "{\"Properties\":{\"GroupName\":\"default\",\"ScheduleExpression\":\"rate(1 minutes)\",\"Target\":{\"Arn\":\"LAMBDA_ARN\",\"RetryPolicy\":{\"MaximumRetryAttempts\":\"100\",\"MaximumEventAgeInSeconds\":\"86400\"},\"RoleArn\":\"ROLE_ARN\"},\"Description\":\"\",\"State\":\"DISABLED\",\"FlexibleTimeWindow\":{\"Mode\":\"false\"},\"ScheduleExpressionTimezone\":\"UTC\",\"Name\":\"RESOURCE_NAME\"}}"
```

Changeset execution failure error:
> Properties validation failed for resource EventBridgeSchedule with message: [#/FlexibleTimeWindow/Mode: false is not a valid enum value]

**Comment by richard-grainger-veeva** [comment]
> I'm seeing the same thing while using a JSON template, and rain v1.23.0. I'm updating an existing stack using `rain` that was previously deployed using the AWS CLI. The `FlexibleTimeWindow` `Mode` property value is being changed from `"OFF"` in my template to `"false"` at deployment.

Try installing from `main`, which has the fix:

```
go install github.com/aws-cloudformation/rain/cmd/rain@main
```

**Comment by leinardi** [comment]
@evilneuro also, try to use the `--node-style strict-booleans` parameter: it fixes the issue for YAML files, maybe works for JSON as well. And this option is already available in v1.23.0.
